parquet/src/main/java/org/apache/iceberg/parquet/Parquet.java

```diff
@@ -1160,8 +1162,37 @@ public <D> CloseableIterable<D> build() {
           optionsBuilder.withDecryption(fileDecryptionProperties);
         }
 
+        boolean pushedFilters = true;
         ParquetReadOptions options = optionsBuilder.build();
 
+        if (filter != null) {
```

Comment:
This is code duplication from below block. We can extract the common logic to form `ParquetFilters`

Reply:
ack